server/api/attendance/attendance.controller.js

```diff
@@ -365,20 +365,56 @@ exports.attend = function(req,res){
           return res.status(200).json({'type':'Full group bonus attendance', 'unverified': needsVerification});
         });
       }
-      // Classyear attendance code and bonus code was incorrect, try small group
       else{
+      // Classyear attendance code and bonus code was incorrect, try small group        
         return SmallGroup.findOne({"students":user._id, "classYear":classYear._id})
     	  .select('+dayCodes.code')
         .exec(function(err, smallgroup){
-          if (err) {return handleError(err)}
-          // if the user has no smallgroup, they cannont submit smallgroup  attendance
+          if (err) {return handleError(err)}          
+          
+          //if the user has no smallgroup,try comparing the code submission with the lastest dayCodes
           if (!smallgroup){
-              return res.status(400).json('No small group found or incorrect daycode!');
+              SmallGroup.findOne({"dayCodes.code":code,"classYear":classYear._id})
```

Comment:
could we break this out into a helper function? it is a lot of logic and code to be embedded in the controller